scripts/admin_redistribute_routers.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python
+"""
+This script will take all users without a 'router_name' credential in Identity and assign it to them.
+Distribution will occur evenly across the list in `provider.get_routers()`, and take into account previously set router_names.
+"""
+import argparse
+import django; django.setup()
+from core.models import Provider, Identity
+
+
+def main(args):
+    provider_id = args.provider
+    redistribute = args.redistribute
+    for provider in Provider.objects.filter(id=provider_id):
+        if redistribute:
+            needs_router = provider.identity_set.all()
+        else:
+            needs_router = provider.missing_routers()
+
+        router_map = provider.get_router_distribution(redistribute=redistribute)  # Print 'before'
+        if redistribute:
+            router_map = {key: 0 for key in router_map.keys()}
+        for identity in needs_router:
+            # Select next available router for the identity
+            selected_router = provider.select_router(router_map)
+            # Save happens here:
+            Identity.update_credential(identity, 'router_name', selected_router, replace=True)
```

Comment:
Note that we only update credentials to actually assign a new router name from an admin script.

We don't want to adversely affect users by assigning a router_name before all their instances have been suspended and their tenant network has been dismantled.

This script provides the ability to redistribute the routers as part of a maintenance period where instances are stopped or suspended.